toolkit/legend/src/main/java/com/arcgismaps/toolkit/legend/Legend.kt

```diff
@@ -219,8 +220,30 @@ public fun Legend(
                     )
                 }
         }
-    }
 
+        // observe any changes to Layer's visibility and rebuild the legend if anything changes.
+        LaunchedEffect(Unit) {
+            val visibilityChanges: List<SharedFlow<Boolean>> =
+                operationalLayers.filterIsInstance<Layer>().map {
+                    it.visibilityChanged
+                } + (basemap?.baseLayers?.map {
+                    it.visibilityChanged
+                } ?: emptyList()) + (basemap?.referenceLayers?.map {
+                    it.visibilityChanged
+                } ?: emptyList())
+
+            merge(*visibilityChanges.toTypedArray()).shareIn(this, SharingStarted.Lazily).collect {
```

Comment:
instead of collecting for every visibility change, this will allow to only collect the latest. 
from the doc: 
> The crucial difference from collect is that when the original flow emits a new value then the `action block` for the previous value is cancelled.
```suggestion
            merge(*visibilityChanges.toTypedArray()).shareIn(this, SharingStarted.Lazily).collectLatest {
```

Reply:
@eri9000 have you tested this?